### PR TITLE
fix gremlin job result size gt cassandra limit and lt hugegraph limit

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: Iade74c66259214a6895ae369374c7243761cb0e0

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskCallable.java [hunk 1]
```diff
@@ -35,8 +35,9 @@
 
     private static final Logger LOG = Log.logger(HugeTask.class);
 
-    private static final String ERROR_MAX_LEN = "Failed to commit changes: " +
-                                                "The max length of bytes is";
+    private static final String ERROR_COMMIT = "Failed to commit changes: ";
+    private static final String ERROR_MAX_LEN = "The max length of bytes is";
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskCallable.java:39]: define Set

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskCallable.java [hunk 2]
```diff
@@ -98,7 +99,15 @@ protected void save() {
                  */
                 LOG.error("Failed to save task with error \"{}\": {}",
                           e, task.asMap(false));
-                if (e.getMessage().contains(ERROR_MAX_LEN)) {
+                String message = e.getMessage();
+                /*
+                 * ERROR_TOO_LARGE_BATCH occurs when using cassandra store
+                 * and result size is in
+                 * [batch_size_fail_threshold_in_kb, TASK_RESULT_SIZE_LIMIT)
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskCallable.java:106]: improve comments

**Review by javeme — CHANGES_REQUESTED** [review]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskCallable.java [hunk 3]
```diff
@@ -117,6 +126,15 @@ protected void save() {
         }
     }
 
+    private static boolean needSaveWithEx(String message) {
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/TaskCallable.java:129]: move to line 167

**Review by javeme — APPROVED** [review]

**Comment by javeme** [comment]
image

**Review by Linary — APPROVED** [review]

**Pull request merged**